### PR TITLE
Routing splitting waypoints

### PR DESCRIPTION
**Pull request by Ulvetanna**
# PolarRoute Pull Request Template
## Description of change
Adding in the functionality to split the mesh aground the waypoints locations to reduce the discrepancy for the route construction. This can now be toggled by an optional parameter within the config file
 
## Fixes # (issue)
#275 

# Testing
To ensure that the functionality of the PolarRoute codebase remains consistent throughout the development cycle a testing strategy has been developed, which can be viewed in the document `test/testing_strategy.md`. 
This includes a collection of test files which should be run according to which part of the codebase has been altered in a pull request. Please consult the testing strategy to determine which tests need to be run. 

- [x] My changes have not altered any of the files listed in the testing strategy

- [ ] My changes result in all required regression tests passing without the need to update test files.  
  
> *list which files have been altered and include a pytest.txt file for each of
> the tests required to be run*
>
> The files which have been changed during this PR can be listed using the command

    git diff --name-only 0.4.x

- [ ] My changes require one or more test files to be updated for all regression tests to pass.   

> *include pytest.txt file showing which tests fail.*  
> *include reasoning as to why your changes cause these tests to fail.* 
>
> Should these changes be valid, relevant test files should be updated.  
> *include pytest.txt file of test passing after test files have been updated.*

# Checklist

- [x] My code follows [pep8](https://peps.python.org/pep-0008/) style guidelines.  
- [x] I have commented my code, particularly in hard-to-understand areas.  
- [ ] I have updated the documentation of the codebase where required.  
- [ ] My changes generate no new warnings.   
- [ ] My PR has been made to the `0.4.x` branch (**DO NOT SUBMIT A PR TO MAIN**)

### CODE REVIEW

**Comment by Ulvetanna** [comment]
Please find attached the testing logs for `test_routes_dijkstra.py` and `test_routes_smoothed.py`. 

[dijkstra_test.log](https://github.com/antarctica/PolarRoute/files/14994776/dijkstra_test.log)
[smoothed_test.log](https://github.com/antarctica/PolarRoute/files/14994774/smoothed_test.log)

Currently it's failing on the smoothed routes and I will investigate further. This could be due to the changes in a hotfix from a while back where the blocking condition changed. I will investigate further

**Review by SamuelHall700 — APPROVED** [review]
Looks good!

**Comment by Ulvetanna** [comment]
Requirements needed to add:
- [ ] Update Test for Smoothed Paths
- [ ] Add in tests for waypoint splitting example

File: polar_route/crossing_smoothing.py [hunk 1]
```diff
@@ -1047,14 +1047,23 @@ def blocked(self,new_cell,cell_a,cell_b):
         end   = cell_b['SIC']
```
Comment (gecoombs) [inline polar_route/crossing_smoothing.py:1047]: Should the docstring here say 5% or 10%?

Comment (Ulvetanna) [reply polar_route/crossing_smoothing.py:1047]: Good point it should actually say something like 'depending on the blocking condition defined in the route config (defaults: 10)'. I will add this as a change with the regression tests coming soon

**Comment by Ulvetanna** [comment]
Updated the regression tests with the corrected blocking conditions. The blocking condition has a weird effect on the `fuel` optimised path. This needs to be investigated further before moving onto the next bit of the pull request

**Pull request merged**